### PR TITLE
Implementation for uploading HTML file sets

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

**Comment by MrCreosote** [comment]
@msneddon do you mind giving this a once over before I merge? It'd be good to get another set of eyes on it.

File: HTMLFileSetUtils.spec [hunk 1]
```diff
@@ -1,7 +1,50 @@
...
+A utility for uploading HTML file sets to the KBase data stores.
 */
 
 module HTMLFileSetUtils {
 
+	/* A HTML file set as saved in the Workspace Service.
+		file - a base64 encoded zip file containing HTML files.
+	 */
+	typedef structure {
+		string file;
```
Comment (msneddon) [inline HTMLFileSetUtils.spec:11]: Seems like we would want this service to abstract away any base64 encoding scheme?  I don't think we should make consumers of this service encode the data.

This parameter name also isn't particularly clear - is this a file path or the actual file content?

Comment (MrCreosote) [reply HTMLFileSetUtils.spec:11]: The service does exactly that. This is the workspace type, where the data is stored, so this is an accurate description of the data. The service API completely encapsulates this type.

It's the actual file, which is why I called it file.

**Comment by MrCreosote** [comment]
Merging on Mike's behalf

**Pull request merged**